### PR TITLE
Fix crash with non vanilla wood

### PR DESCRIPTION
**Pull request by Volune**
This fix a crash if AgriCraft is used with another mod that add new wood planks.

### CODE REVIEW

File: src/main/java/com/infinityraider/agricraft/utility/CustomWoodType.java [hunk 1]
```diff
@@ -86,8 +78,17 @@ public ItemStack getStack() {
     }
 
     @SideOnly(Side.CLIENT)
-    public ResourceLocation getTexture() {
-        return texture;
+    public TextureAtlasSprite getIcon() {
+        try {
```
Comment (InfinityRaider) [inline src/main/java/com/infinityraider/agricraft/utility/CustomWoodType.java]: put the try {} catch() inside the if(texture == null), and in the catch statement, assign the missing sprite.

File: src/main/java/com/infinityraider/agricraft/blocks/irrigation/BlockSprinkler.java [hunk 2]
```diff
@@ -141,10 +140,10 @@ public boolean canPlaceBlockAt(World world, BlockPos pos) {
 	}
 
 	@SideOnly(Side.CLIENT)
-	public ResourceLocation getIcon(IBlockAccess world, BlockPos pos, IBlockState state, EnumFacing side, @Nullable TileEntityBase te) {
+	public TextureAtlasSprite getIcon(IBlockAccess world, BlockPos pos, IBlockState state, EnumFacing side, @Nullable TileEntityBase te) {
 		TileEntity channel = world.getTileEntity(pos.add(0, 1, 0));
 		if (channel != null && channel instanceof TileEntityChannel) {
-			return ((TileEntityChannel) channel).getTexture();
+			return ((TileEntityChannel) channel).getIcon();
```
Comment (InfinityRaider) [inline src/main/java/com/infinityraider/agricraft/blocks/irrigation/BlockSprinkler.java]: Don't return null icons please, return missing sprite instead.

**Comment by Volune** [comment]
Should be ok now.

By the way, I've seen a crash when breaking some of the AgriCraft blocks. Doesn't seem related with this issue.

**Comment by InfinityRaider** [comment]
Probably the multiblock code, multiblock code is bugged since the 1.8+ port, I think its the last thing to finish this port.

**Pull request merged**